src/sidebar/services/router.js

```diff
@@ -61,13 +61,31 @@ export default function router($window, store) {
     return url;
   }
 
+  let didRegisterPopstateListener = false;
+
   /**
    * Synchronize the route name and parameters in the store with the current
    * URL.
+   *
+   * The first call to this method also registers a listener for future back/forwards
+   * navigation in the browser.
    */
   function sync() {
     const { route, params } = currentRoute();
     store.changeRoute(route, params);
+
+    // Setup listener for back/forward navigation. We do this in `sync()` to avoid
+    // the route being changed by eg. a "popstate" emitted by the browser on
```

Comment:
You say "e.g." here, but could this happen by any other mechanism (than `popstate`)?

Reply:
Oh, I meant "eg." as in "this is an example of when `popstate` might be emitted before `sync()` is called".